app/src/main/java/com/dessalines/thumbkey/utils/Utils.kt

```diff
@@ -747,6 +747,20 @@ fun autoCapitalizeCheck(ime: IMEService): Boolean {
     // Knows if its an empty field
     val empty = ime.currentInputConnection.getTextBeforeCursor(1, 0).isNullOrEmpty()
 
+    // Avoid capitalizing in certain edit boxes
+    val inputType = ime.currentInputEditorInfo.inputType and (InputType.TYPE_MASK_VARIATION)
+    if (listOf(
+            InputType.TYPE_TEXT_VARIATION_URI,
+            InputType.TYPE_TEXT_VARIATION_PASSWORD,
+            InputType.TYPE_TEXT_VARIATION_WEB_PASSWORD,
+            InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD,
+        ).contains(inputType)
+    ) {
+        return false
+    }
+    // TYPE_NULL indicates no automation desired. Notably fixes termux
+    if (ime.currentInputEditorInfo.inputType == EditorInfo.TYPE_NULL) return false
```

Comment:
Might as well move this condition into the check clause above, since they have the same end-result.